### ISSUE TITLE
[FEATURE] Does not work with the "links at the end format"

### ISSUE THREAD

**Issue by XanatosX** [treymo/joplin-link-graph#67]
It seems like this plugin does not work correctly with links defined at the end of the markdown as seen here

![grafik](https://github.com/treymo/joplin-link-graph/assets/10531466/d9a813b8-7f78-4540-a863-85fb0c5d9eff)

Since this is common practice for markdown I would like to see this working as well.

**Comment by Petr05** [comment]
Agreed! Most of my Joplin notes are linked via ['reference link'][Ref1] formatting (this example includes a URL rather than a note ID). Sadly, with this particular note-linking syntax unaccounted for, the graph is essentially ineffective for my particular use.

[Ref1]:https://daringfireball.net/projects/markdown/syntax#link

**Comment by treymo** [comment]
@XanatosX what do you mean by "links defined at the end of the markdown"- I don't quite understand.   In the preview pane of your note, it doesn't look like it's rendering anything under "Fleisch".  That makes me think Joplin doesn't even understand that link format.

@Petr05 Joplin does not appear to support links of the "reference link" format: `[<link text>]: <link destination>`. Are you able to products links in Joplin that are clickable with that format? (Ignoring this graph plugin for now)

**Comment by XanatosX** [comment]
I did create a small test file to confirm that this is indeed working.

![image](https://github.com/treymo/joplin-link-graph/assets/10531466/231d9722-8e82-44f0-97fa-6db4205dd4f8)
![image](https://github.com/treymo/joplin-link-graph/assets/10531466/f9e9e431-a0ab-41c3-b2b6-e579ea63b1d4)

Both links are clickable and do work.

I do use the following Joplin configuration

```
Joplin 2.13.13 (prod, win32)

Client ID: 1af169ac80bd4ef1ae58b10185346988
Sync Version: 3
Profile Version: 44
Keychain Supported: Yes

Revision: c3cdeeb

Link Graph UI: 1.5.0
MultiMarkdown Table Tools: 1.1.0
Note Link System: 0.8.0
Publish to Web: 1.1.0
Templates: 2.4.0
```

So Joplin can work with the provided format, at least on a windows machine with the plugins listed above. Hopefully that information is helpful.

**Comment by treymo** [comment]
Ahhh, I see, I didn't quite understand the format.  I was able to confirm that Joplin does indeed work with those.

This would definitely be possible, would just require changes to how we parse links from an individual note. So basically just a second regex in the existing `getAllLinksForNote` function.  No clue when I would be able to get to this but should be doable

**Comment by Petr05** [comment]
Spot on, ["reference link"][ref-link] is the formatting in question so we're all on the same page here. If/when the plugin becomes compatible with it, I'll be very grateful! Also grateful, of course, for the effort you've already invested.  

[ref-link]: https://daringfireball.net/projects/markdown/syntax#link

**Comment by Petr05** [comment]
Also, I've re-opened this same issue beyond the scope of joplin-link-graph: https://discourse.joplinapp.org/t/links-to-other-notes-in-exported-html
